## Introduction

### Objective
The primary objective of this analysis is to examine the historical performance of Chester, Inc., as well as its market competitors. Through a comprehensive examination of various datasets, we aim to extract actionable insights that could pave the way for strategic decisions and future growth.

---

### Scope
The analysis covers multiple facets of business performance, including but not limited to:
- Sales and Revenue Trends
- Market Share Analysis
- Human Resource Metrics
- Financial Ratios and Health

---

### Methodology
We'll employ Python's data manipulation and visualization libraries—Pandas, Matplotlib, and Seaborn—to perform this analysis. The study is organized into five phases:
1. **Data Preparation**: Importing libraries and loading datasets.
2. **Exploratory Data Analysis (EDA)**: Summary statistics and correlation analysis.
3. **Data Visualization**: Creating insightful charts and graphs.
4. **Insights and Recommendations**: Drawing conclusions and suggesting actionable steps.
5. **Conclusion**: Summarizing the key takeaways.

---

### Importance
The insights derived from this analysis will not only provide a snapshot of the company's current state but also shape its future direction. This is critical for stakeholders in making informed decisions.


## Phase 1:  Data Preparation

### Step 1: Import libraries

First, we'll import the Python libraries that will be needed for data manipulation and visualization.

In [2]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

### Step 2: Load the Data

Next, we will load the data from the Excel file into Python.

In [3]:
# Read the Excel file data into a dictionary of dataframes
excel_file = '../../data/raw/CompetitionRoundsData.xlsx'
sheets_dict = pd.read_excel(excel_file, sheet_name=None)

### Step 3: Data Review

Now we will utilize various Pandas methods and functions to familiarize ourselves with the data.

First let's see what data is available:

In [4]:
# Get the list of worksheet names
worksheet_names = list(sheets_dict.keys())

# Display the worksheet names in a user-friendly manner
print('Available Worksheets:')
for index, sheet in enumerate(worksheet_names):
    print(f'{index+1}. {sheet}')

Available Worksheets:
1. Years
2. Forecast
3. selectedFinancials
4. Stocks
5. Bonds
6. Financial
7. ProductInfo
8. Size Stats
9. Customer Criteria
10. ProductBySegment
11. MarketShare
12. actualMarketShareUnits
13. potentialMarketShareUnits
14. actualMarketSharePercent
15. potentialMarketSharePercent
16. HR
17. TQM
18. Ethics
19. officialDecMrkt
20. officialDecPrd
21. officialDecHR
22. officialDecFin


Wow, that is a lot of data!

Let's review further with a systematic approach:
1. Check for Missing Values, Number of Numeric/Categorical Columns
2. Data Shape/Types
3. Summary Statistics
4. View Sample Rows
5. Drop Unwanted Sheets

#### Step 3.1: Basic Info

We'll start by looping through the dictionary and print out basic information for each DataFrame:

In [5]:
# Initialize an empty DataFrame to store summary
summary_df = pd.DataFrame(columns=['Sheet Name', 'Missing Values', 'Numeric Cols', 'Categorical Cols'])

# Initialize an index counter
index_counter = 0

# Iterate through sheets_dict to populate summary_df
for name, df in sheets_dict.items():
    missing_values = df.isnull().sum().sum() | df.isna().sum().sum()
    numeric_cols = len(df.select_dtypes(include=['number']).columns)
    categorical_cols = len(df.select_dtypes(include=['object']).columns)

    summary_df.loc[index_counter] = [name, missing_values, numeric_cols, categorical_cols]
    index_counter += 1

# Set the index of summary_df to Sheet Name for better readability
summary_df.set_index('Sheet Name', inplace=True)

# Display the summary DataFrame
print(summary_df)

                             Missing Values  Numeric Cols  Categorical Cols
Sheet Name                                                                 
Years                                     0             1                 2
Forecast                                  0            13                 2
selectedFinancials                        0            13                 1
Stocks                                    0            10                 1
Bonds                                     0             4                 3
Financial                                 0            47                 1
ProductInfo                               0            15                 3
Size Stats                                0             5                 2
Customer Criteria                         0             3                 4
ProductBySegment                        158            13                 4
MarketShare                            1904            13                 2
actualMarket

#### Step 3.2: Data Shape/Types

In [6]:
# Used .info() to get a concise summary of each DataFrame
for name, df in sheets_dict.items():
    print(f"Sheet Name: {name}")
    print(f"Info:")
    print(f"{df.info()}")
    print("-"*50)

Sheet Name: Years
Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9 entries, 0 to 8
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   round        9 non-null      int64         
 1   yearEnding   9 non-null      datetime64[ns]
 2   companyName  9 non-null      object        
 3   companyID    9 non-null      object        
dtypes: datetime64[ns](1), int64(1), object(2)
memory usage: 420.0+ bytes
None
--------------------------------------------------
Sheet Name: Forecast
Info:
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 15 columns):
 #   Column                                                                           Non-Null Count  Dtype  
---  ------                                                                           --------------  -----  
 0   Round                                                                            20 non-null     

#### Step 3.3: Summary Statistics

In [8]:
# Utilize the describe() method to get a statistical summary of each DataFrame
for name, df in sheets_dict.items():
    print(f"Sheet Name: {name}")
    print(f"Statistical Summary:")
    print(f"{df.describe()}")
    print("-"*50)

Sheet Name: Years
Statistical Summary:
          round           yearEnding
count  9.000000                    9
mean   4.000000  2027-12-31 08:00:00
min    0.000000  2023-12-31 00:00:00
25%    2.000000  2025-12-31 00:00:00
50%    4.000000  2027-12-31 00:00:00
75%    6.000000  2029-12-31 00:00:00
max    8.000000  2031-12-31 00:00:00
std    2.738613                  NaN
--------------------------------------------------
Sheet Name: Forecast
Statistical Summary:
           Round  Total Industry Unit Demand  Next Year's Segment Growth Rate  \
count  20.000000                   20.000000                        20.000000   
mean    1.500000                 5362.650000                         0.150400   
std     1.147079                 3544.408885                         0.041011   
min     0.000000                 1915.000000                         0.092000   
25%     0.750000                 2554.000000                         0.117000   
50%     1.500000                 3370.500000     

Here are my observations:

- ~~Years: 9 rows, 4 cols, no missing values, 1 numeric col, 2 cat cols....~~ NOT using for analysis
- ~~Forecast: 20 rows, 15 cols, no missing values, 13 numeric cols, 2 cat cols....~~ NOT using for analysis
- selectedFinancials: 54 rows, 15 cols, no missing values, 13 numeric cols, 1 cat col....*interested in using*
    - ROS, ROA, ROE, Sales, EBIT, Profits, cumulativeProfits, SG&A/Sales, contribMarginPercent
- Stocks: 54 rows, 12 cols, no missing values, 10 numeric cols, 1 cat col....*interested in using*
    - Close, Change, EPS
- ~~Bonds: 131 rows, 8 cols, 4 numberic cols, 3 cat cols...~~ NOT using for analysis
- Financial: 54 rows, 49 cols, no missing values, 47 numeric cols, 1 cat col....*interested in using*
    - Net Income (Loss), Net cash from operations, Plant improvements (net), Net cash from financing activities, Net change in cash position, Cash, Accounts Receivable, Inventory, Total Current Assets, Total Fixed Assets, Total Assets, Accounts Payable, Total Current Liabilities, Total Liabilities, Total Equity, Total Liabilities & Owners' Equity, Sales, Variable Costs (Labor, Material, Carry), Contribution Margin, SGA (R&D, Promo, Sales, Admin), Other (Fees, writeoffs, TQM, Bonuses), EBIT, Interest (Short term, Long term), Taxes, Profit Sharing, Net Proft
- ProductInfo: 284 rows, 20 cols, no missing values, 15 numeric cols, 3 cat cols...*interested in using*
    - product, segment, unitsSold, Inventory, ageDec31, mtbf, pfmn, size, price, materialCost, laborCost, contrMargin, automationNR,  plantUtiliz
- Size Stats: 45 rows, 8 cols, no missing values, 5 numeric cols, 2 cat cols....*interested in using*
    - Total Industry Unit Demand, Actual Industry Unit Sales, Segment % of Total Industry, Next Year's Segment Growth Rate
- Customer Criteria: 180 rows, 8 cols, no missing values, 3 numeric cols, 4 cat cols....*interested in using*
    - criteria, Expectations (need to delete redundant text and split into low/high and Pfmn/Size), Importance
- ProductBySegment: 332 rows, 19 cols, 158 missing values in stockOut col, 13 numeric cols, 4 cat cols....*interested in using*
    - marketShare, stockOut~~(fill missing values with 0)~~, promoBudget, custAwareness, salesBudget, custaccessibility, custSurveyDec
- ~~MarketShare: 306 rows, 16 cols, 1904 missing values, 13 numeric cols, 2 cat cols....~~ NOT using for analysis
- actualMarketShareUnits: 9 rows, 9 cols, no missing values, 7 numeric cols, 1 cat col....*interested in using*
    - Industry Unit Sales for Trad Seg, Industry Unit Sales for Low Seg, Industry Unit Sales for High Seg, Industry Unit Sales for Pfmn Seg, Industry Unit Sales for Size Seg
- potentialMarketShareUnits: 9 rows, 9 cols, no missing values, 7 numeric cols, 1 cat col....*interested in using*
    - Units Demanded for Trad Seg, Units Demanded for Low Seg, Units Demanded for High Seg, Units Demanded for Pfmn Seg, Units Demanded for Size Seg
- actualMarketSharePercent: 9 rows, 9 cols, no missing values, 7 numeric cols, 1 cat col....*interested in using*
    - Actual % of Market for Trad Seg, Actual % of Market for Low Seg, Actual % of Market for High Seg, Actual % of Market for Pfmn Seg, Actual % of Market for Size Seg
- potentialMarketSharepercent: 9 rows, 9 cols, no missing values, 7 numeric cols, 1 cat col....*interested in using*
    - Potential % of Market for Trad Seg, Potential % of Market for Low Seg, Potential % of Market for High Seg, Potential % of Market for Pfmn Seg, Potential % of Market for Size Seg
- HR: 54 rows, 23 cols, no missing values, 20 numeric cols, 2 cat cols....*interested in using*
    - Overtime%, Turnover Rate, New Employees, Productivity Index, Recruiting Cost, Separation Cost, Training Cost, Total HR Admin Cost, Wages, Annual Raise
- TQM: 54 rows, 19 cols, no missing values, 17 numeric cols, 1 cat col....*interested in using*
    - Total Expenditures, Material Cost Reduction, Labor Cost Reduction, Reduction R&D Cycle Time, Reduction Admin Costs, Demand Increase
- ~~Ethics: 63 rows, 10 cols, no missing values, 8 numeric cols, 1 cat col....~~ NOT using for analysis
- officialDecMrkt: 45 rows, 16 cols, no missing values, 11 numeric cols, 4 cat cols....*interested in using*
    - MKTGDirectMail, MKTGDistributors, MKTGEmail, MKTGInsideSales, MKTGOutsideSales, MKTGPrintMedia, MKTGReports, MKTGTradeShows, MKTGWebMedia
- ~~officialDecPrd: 56 rows, 15 cols, no missing values, 11 numeric cols, 3 cat cols....~~ NOT using for analysis
- ~~officialDecHR: 14 rows, 6 cols, no missing values, 4 numeric cols, 1 cat col....~~ NOT using for analysis
- ~~officialDecFin: 8 rows, 12 cols, no missing values, 9 numeric cols, 2 cat cols....~~ NOT using for analysis

#### Step 3.4: Peek into Data

Taking a peek into the actual data by utilizing the sample() method and confirming agreement with existing observations.


In [10]:
# Display a data sampling of each DataFrame
for name, df in sheets_dict.items():
    print(f"Sheet Name: {name}")
    print(f"Data Sample:")
    print(f"{df.head()}")
    print("-"*50)

Sheet Name: Years
Data Sample:
   round yearEnding companyName companyID
0      0 2023-12-31     Chester   C142507
1      1 2024-12-31     Chester   C142507
2      2 2025-12-31     Chester   C142507
3      3 2026-12-31     Chester   C142507
4      4 2027-12-31     Chester   C142507
--------------------------------------------------
Sheet Name: Forecast
Data Sample:
   Round Segment  Company  Total Industry Unit Demand  \
0      3    Trad  Chester                        9619   
1      3     Low  Chester                       12488   
2      3    High  Chester                        4007   
3      3    Perf  Chester                        3293   
4      3    Size  Chester                        3284   

   Next Year's Segment Growth Rate  Next Year's Segment Demand  \
0                            0.092                     884.948   
1                            0.117                    1461.096   
2                            0.162                     649.134   
3                        

#### Step 3.5: Drop Unwanted Sheets

In [11]:
# Drop specific sheets (rows in this DataFrame) that are not the focus of this analysis
sheets_to_drop = ['Years', 'Forecast', 'Bonds', 'MarketShare', 'Ethics', 'officialDecPrd', 'officialDecHR', 'officialDecFin']

# Dropping the rows
summary_df.drop(sheets_to_drop, inplace=True)

# Displaying the updated summary DataFrame
print("\nUpdated summary DataFrame after dropping specific sheets:")
print(summary_df)


Updated summary DataFrame after dropping specific sheets:
                             Missing Values  Numeric Cols  Categorical Cols
Sheet Name                                                                 
selectedFinancials                        0            13                 1
Stocks                                    0            10                 1
Financial                                 0            47                 1
ProductInfo                               0            15                 3
Size Stats                                0             5                 2
Customer Criteria                         0             3                 4
ProductBySegment                        158            13                 4
actualMarketShareUnits                    0             7                 1
potentialMarketShareUnits                 0             7                 1
actualMarketSharePercent                  0             7                 1
potentialMarketSharePercent  

### Step 4: Data Cleaning

#### Step 4.1 Handle Missing Values

In [ ]:
# Access the DataFrame from sheets_dict
product_by_segment_df = sheets_dict['ProductBySegment']

# Fill NaN values in the 'stockOut' column with 0
product_by_segment_df['stockOut'].fillna(0, inplace=True)

# Update the DataFrame in sheets_dict
sheets_dict['ProductBySegment'] = product_by_segment_df

# Update the 'Missing Values' count for 'ProductBySegment' in summary_df
summary_df.loc['ProductBySegment', 'Missing Values'] = 0

# Displaying the updated summary DataFrame
print("\nUpdated summary DataFrame after updating 'Missing Value' count for 'ProductBySegment':")
print(summary_df)

#### Step 4.2 Splitting Columns/Deleting Redundant Text

In [ ]:
customer_criteria_df = sheets_dict['Customer Criteria']

# Extract 'Ideal Age'
customer_criteria_df['Ideal Age'] = customer_criteria_df['Expectations'].str.extract('Ideal Age = ([\d.]+)').astype(float)

# Extract 'Pfmn'
customer_criteria_df['Pfmn'] = customer_criteria_df['Expectations'].str.extract('Pfmn ([\d.]+)').astype(float)

# Extract 'Size'
customer_criteria_df['Size'] = customer_criteria_df['Expectations'].str.extract('Size ([\d.]+)').astype(float)

# Extract 'Low Price'
customer_criteria_df['Low Price'] = customer_criteria_df['Expectations'].str.extract('\$([\d\d.]+) ').astype(float)

# Extract 'High Price'
customer_criteria_df['High Price'] = customer_criteria_df['Expectations'].str.extract('- ([\d.]+)').astype(float)

# Extract 'MTBF Low'
customer_criteria_df['MTBF Low'] = customer_criteria_df['Expectations'].str.extract('MTBF ([\d.]+)').astype(float)

# Extract 'MTBF High'
customer_criteria_df['MTBF High'] = customer_criteria_df['Expectations'].str.extract('-([\d.]+)').astype(float)

# Drop the 'Expectations' column
customer_criteria_df.drop('Expectations', axis=1, inplace=True)

# Display the updated DataFrame
print(customer_criteria_df.head())

In [ ]:
# Replace values in 'criteria' column
customer_criteria_df['criteria'].replace({
    'Age': 'Ideal Age',
    'Reliability': 'Reliability (MTBF)'
}, inplace=True)

# Display the updated DataFrame
print(customer_criteria_df.head())